xds/internal/balancer/clusterresolver/configbuilder.go

```diff
@@ -190,7 +190,18 @@ func buildClusterImplConfigForEDS(g *nameGenerator, edsResp xdsresource.Endpoint
 		})
 	}
 
-	priorities := groupLocalitiesByPriority(edsResp.Localities)
+	var priorities [][]xdsresource.Locality
+	// Triggered by an NACK or resource-not-found error before update, or a
+	// empty localities list in a update. In either case want to create a
+	// priority, and send down empty address list, causing TF for that priority.
+	if len(edsResp.Localities) == 0 {
+		// "If any discovery mechanism instance experiences an error retrieving
+		// data, and it has not previously reported any results, it should
+		// report a result that is a single priority with no endpoints." - A37
+		priorities = [][]xdsresource.Locality{{}}
+	} else {
+		priorities = groupLocalitiesByPriority(edsResp.Localities)
+	}
```

Comment:
Minor nit. You can shorten this even further:

```golang
priorities := [][]xdsresource.Locality{{}}
if len(edsResp.Localities) != 0 {
	priorities = groupLocalitiesByPriority(edsResp.Localities)
}
```

Reply:
Thanks for suggestion. Switched.